### ISSUE TITLE
Wizard Form - Issue approving form when one user has two tasks within a role

### ISSUE THREAD

**Issue by mwojno-tsg** [tsgrp/HPI#1974]
Overview
- There is currently an issue when a user has two tasks within a role. When they go to approve the first task, it will update the approval details that both tasks are approved. However, the Complete Wizard Approval folder action is still active. When, they try to approve it again there will be a 500 error on the approval.

Reproduce
- To reproduce. Create a form. Make sure there is a proxy to a user who already has a task or a future reassigned task to a user that already is one of the approvers. This will make sure one user can have two tasks. 
- EX: I have a proxy from cobrien -> qdadmin2, cobrien is an approver in CR Approval; qdadmin2 is also an approver in CR Approval

![2019-10-16 11_13_07-](https://user-images.githubusercontent.com/54864662/66938744-5f8ca180-f007-11e9-8b10-45712475085b.png)

- Approve the form once for qdadmin2 (or the user with two tasks). 
- Check the approval details and you should see the qdadmin2 has been approved for both tasks not the first task, which should be the case.

![2019-10-16 11_15_00-](https://user-images.githubusercontent.com/54864662/66938795-77642580-f007-11e9-8b91-19c475787181.png)

- Also, you can see that the Complete Wizard Approval folder action is still active even though the approval details is saying qdadmin2 doesn't have anymore tasks. This essentially causes the error we will see below.
- Now go and try to approve the second task that qdadmin2 has and you get a 500 error.

![2019-10-16 11_15_38-](https://user-images.githubusercontent.com/54864662/66938908-a4183d00-f007-11e9-95fc-df5ba242d776.png)

Ideas about the fix
- The error should be originating in the CompleteUserTaskListener. I would expect that it is looping through and updating the approval details to say that the user (with two tasks) just approved so update them in the approval details to approved. 
- This would cause the conflict because a task is still outstanding, even though the approval details says otherwise.
- So, a fix would have to be made to only update that task's approval detail signature even though it could be the same user.

**Comment by astocks7** [comment]
This is fixed by breaking out of the loop in CompleteUserTaskListener to ensure we only ever complete one task at a time. Ran into this same issue when implementing group route for review, and we fixed it the same way there. Will get the fix pushed up next week when people are back.

**Comment by astocks7** [comment]
WIZARD FORM APPROVAL - Currently when a user has two tasks within the same role (possible via a proxy), attempting to approve one of the tasks results in approving both which causes an error when trying to approve the (already approved) second task. This is fixed by breaking out of the completed signatures loop in CompleteUserTaskListener.java so that we only ever complete one task at a time. A similar approach was taken when group review route was implemented; revision 24178 -- in CompleteReviewTaskListener we added a break to prevent a similar issue from occurring.

OC rev 24212
CR: dgrumieaux, nwoodbine
